script/tool/lib/src/create_all_plugins_app_command.dart

```diff
@@ -259,4 +268,75 @@ dev_dependencies:${_pubspecMapString(pubspec.devDependencies)}
 
     return buffer.toString();
   }
+
+  Future<int> _genNativeBuildFiles() async {
+    // Only run on macOS.
+    // Other platforms don't need generation of additional files.
+    if (!io.Platform.isMacOS) {
+      return 0;
+    }
+
+    final io.ProcessResult result = io.Process.runSync(
+      flutterCommand,
+      <String>[
+        'pub',
+        'get',
+      ],
+      workingDirectory: _appDirectory.path,
+    );
+
+    print(result.stdout);
+    print(result.stderr);
+    return result.exitCode;
+  }
+
+  Future<void> _updateMacosPodfile() async {
+    // Only change the macOS deployment target if the host platform is macOS.
+    if (!io.Platform.isMacOS) {
+      return;
+    }
+
+    final File podfileFile =
+        app.platformDirectory(FlutterPlatform.macos).childFile('Podfile');
+    if (!podfileFile.existsSync()) {
+      throw ToolExit(64);
```

Comment:
I wonder why `64` is the ToolExit code. Could that be documented here?

Reply:
Initially it was suggested to me, to look at the implementation of `_updateAppGradle` to see how changes to project configuration files could be made (see [this comment](https://github.com/flutter/plugins/pull/6517#pullrequestreview-1132219961)).
> This would be done in `script/tool/lib/src/create_all_plugins_app_command.dart`. See `_updateAppGradle` for an example of making a similar change to some Android values.

There, this (to me arbitrary) exit code `64` was thrown in case the relevant android config file didn't exist.
I don't know the meaning of this value either. Looking back at git history, the value was introduced two years ago in https://github.com/flutter/plugins/pull/3544.
The value itself seems to date back even further tho, since the PR mentioned above only moved some files around.

Reply:
This does look it was arbitrary. The way we're handling error codes for any current code is to [define file-level constants starting with 3](https://github.com/flutter/plugins/blob/419bea0c0cfde28b6852f04932aec8e69998b4f4/script/tool/lib/src/format_command.dart#L30-L34) (since 1 and 2 are global errors) for each failure, and using those, so that it's self-documenting. Let's do that here; it's fine that it's inconsistent with the older code in the file since those values aren't meaningful anyway.